File: Logic_HandsOn/Java/Problems on Array,Pattern etc/Program75.java
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Accept N numbers from user and accept one another number as NO , return frequency of NO form it.
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


import java.util.*;

class Solution
{
	public int iSize,arr[],iCnt,Count,iNum;

	public Solution(int iNo,int iNum)
	{
		this.iCnt = 0;
		this.iNum=iNum;
		this.Count=0;
		this.iSize = iNo;
		arr = new int[iSize];
	}

	public void Accept()
	{
		Scanner sObj = new Scanner(System.in);

		System.out.println("Enter "+arr.length+" elements: ");
		for(iCnt=0;iCnt<arr.length;iCnt++)
		{
			System.out.print("Enter "+ (iCnt+1) + "element: ");
			arr[iCnt] = sObj.nextInt();
		}
	}	

	public int CountNo()
	{
		for(iCnt=0;iCnt<arr.length;iCnt++)
		{
			if(arr[iCnt]==iNum)
			{
				Count++;
			}
		}

		return Count;
	}

}

class Program
{
	public static void main(String[] args)
	{
		int iRet = 0;
		Scanner sObj = new Scanner(System.in);

		System.out.println("Enter number of elements:");
		int iLength = sObj.nextInt();

		System.out.println("Enter the number :");
		int iNo = sObj.nextInt();

		Solution obj = new Solution(iLength,iNo);
		obj.Accept();
		iRet = obj.CountNo();
	
		System.out.println("count is: "+iRet);
	}
}

/*
OUTPUT:
Enter number of elements:
6
Enter the number :
66
Enter 6 elements: 
Enter 1element: 85
Enter 2element: 66
Enter 3element: 3
Enter 4element: 66
Enter 5element: 93
Enter 6element: 88
count is: 2

*/

File: Logic_HandsOn/C++/Problems on Digits,String,Array,Pointers etc/program75.cpp
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Accept N numbers from user and accept one another number as NO , return frequency of NO form it.
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


#include<iostream>
using namespace std;

class Solution
{
	private:
		int *iArr;
		int iSize,iCnt,count,iVal;

	public:
		Solution(int iNo,int iVal,int iCnt=0,int count=0)
		{
			this->iVal = iVal;
			this->iSize=iNo;
			iArr = new int[iSize];
		}

		void Accept()
		{
			cout<<"Enter "<<iSize<<" elements:\n";
			for(iCnt=0;iCnt<iSize;iCnt++)
			{
				cout<<"Enter "<<iCnt+1<<" element: ";
				cin>>iArr[iCnt];
			}
		}
		
		int Count()
		{
			for(iCnt=0;iCnt<iSize;iCnt++)
			{
				if(iArr[iCnt]==iVal)
				{
					count++;
				}
				
			}

			return count;
		}

		~Solution()
		{
			delete []iArr;
		}

};

int main()
{
	int iLength=0,iRet=0,iNum=0;

	cout<<"Enter number of elements: ";
	cin>>iLength;
	
	cout<<"Enter number: ";
	cin>>iNum;

	Solution *dObj = new Solution(iLength,iNum);

	dObj->Accept();
	iRet = dObj->Count();

	cout<<"Count is: "<<iRet<<"\n";

	delete dObj;
	return 0;
}

/*
OUTPUT:
Enter number of elements: 6
Enter number: 66
Enter 6 elements:
Enter 1 element: 85
Enter 2 element: 66
Enter 3 element: 3
Enter 4 element: 66
Enter 5 element: 93
Enter 6 element: 88
Count is: 2

*/
